lib/deflate.js

```diff
@@ -143,6 +144,7 @@ function Deflate(options) {
   this.msg    = '';     // error message
   this.ended  = false;  // used to avoid multiple onEnd() calls
   this.chunks = [];     // chunks of compressed data
+  this._dict_set = false; // have we set the dictionary
```

Comment:
I use it below to ensure we are only calling `setHeader`on the first `push`, or am I misunderstanding sth?